plugins/videobasedtracker/com_osvr_VideoBasedHMDTracker.cpp

```diff
@@ -539,11 +574,63 @@ class ConfiguredDeviceConstructor {
     // Using `get` here instead of `[]` lets us provide a default value.
     int cameraID = root.get("cameraID", 0).asInt();
     bool showDebug = root.get("showDebug", false).asBool();
+    size_t solveIterations = root.get("solveIterations", 0).asInt();
+    double maxReprojectionAxisError = root.get("maxReprojectionAxisError", 0).asDouble();
+
+    // Default is an empty list, which can be filled in from the config
+    // file if we find entries there.
+    osvr::vbtracker::SensorDescriptionList mySensors;
+    if (root.isMember("sensors")) {
+      Json::Value sensors = root["sensors"];
+      for (Json::ArrayIndex i = 0; i < sensors.size(); ++i) {
+        if ( !sensors[i].isMember("patterns") ||
+             !sensors[i].isMember("positions") ) {
+          std::cerr << "com_osvr_VideoBasedHMDTracker: Expected patterns "
+            "and positions in sensor description for sensor" << i << std::endl;
+          return OSVR_RETURN_FAILURE;
+        }
+
+        Json::Value patterns = sensors[i]["patterns"];
+        Json::Value positions = sensors[i]["positions"];
+        if (patterns.size() != positions.size()) {
+          std::cerr << "com_osvr_VideoBasedHMDTracker: Mismatched pattern count ("
+            << patterns.size() << ") and position count ("
+            << positions.size() << ") in sensor description for sensor"
+            << i << std::endl;
+          return OSVR_RETURN_FAILURE;
+        }
```

Comment:
Can the schema be reorganized to be (effectively) "an array of structs" (position and pattern pairs, though doesn't have to be an object-style thing, could just be a tuple-style array of two elements) instead of "struct of arrays"? would eliminate this class of errors and simplify this code.